libcxx/include/__tree

```diff
@@ -1276,7 +1310,7 @@ private:
       return nullptr;
     }
 
-    __assign_value(__dest->__value_, __src->__value_);
```

Comment:
As we no longer expose the member directly, all accesses need to go through the member function so the right type is used.

otherwise, every access  needs to be updated to use ifndef CXX03